Biz/Model/RealmOwner.pm

```diff
@@ -347,6 +353,39 @@ sub validate_login {
     return 'NOT_FOUND';
 }
 
+sub validate_password {
+    my($self, $clear_text) = @_;
+    my($canonical_clear_text) = _canonicalize_for_weak_password($clear_text);
+    return 'WEAK_PASSWORD'
+        if $canonical_clear_text eq _canonicalize_for_weak_password($self->get('display_name'))
+        || $canonical_clear_text eq _canonicalize_for_weak_password($self->get('name'))
+        || $canonical_clear_text eq $self->get('realm_id')
+        || _similar_to_email($self, $canonical_clear_text);
+    return;
+}
+
+sub _canonicalize_for_weak_password {
+    my($value) = @_;
+    $value =~ s/\W+//g;
```

Comment:
I think this should be `s/[^a-z0-9]//ig`. Underscore shouldn't be a distinguishing character.